R/stat-binhex.r

```diff
@@ -88,12 +89,30 @@ hexBin <- function(x, y, binwidth) {
   hb <- hexbin(
     x, xbnds = xbnds, xbins = xbins,
     y, ybnds = ybnds, shape = ybins / xbins,
+    IDs = !is.null(weight),
   )
+  
+  if (!is.null(weight)) {
+      ## we need to produce weights for each cell rather than counts. hb@cID contains the
+      ## ID of the cell for each input point and hb@cell contains the vector of cell IDs
+      ## which relate to values in hb@count.
+      
+      ## consequently we can put cID next to weight, and then sum them grouped by cID to get
+      ## total weight for each bin
+      sumWeights <- data.frame(cellID = hb@cID, weight=weight)
+      sumWeights <- aggregate(. ~ cellID, data=sumWeights, FUN=sum)
```

Comment:
I think you could do this without an intermediate data frame by using `tapply()`